cmd/testpeers.go

```diff
@@ -203,6 +202,7 @@ func pingPeerOnce(ctx context.Context, tcpNode host.Host, peer p2p.Peer) (ping.R
 }
 
 func pingPeerContinuously(ctx context.Context, tcpNode host.Host, peer p2p.Peer, resCh chan ping.Result) {
+	defer close(resCh)
```

Comment:
This change alone does not solve the problem of the blocking write. I know this was introduced previously, not in this PR, but this is something I overlooked in past..